src/region/RegionCreator.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from 'react';
+import { Rect } from '../@types';
+
+type Position = {
+    x: number;
+    y: number;
+};
+
+type Props = {
+    canDraw: boolean;
+    className?: string;
+    onDraw: (rect: Rect) => void;
+    onDone: () => void;
+};
+
+type State = {
+    isDrawing: boolean;
+    position: Position | null;
+};
+
+const LIMIT_POS = 1; // Minimum position must be positive
+const LIMIT_SIZE = 10; // Minimum region must be clickable
+const PRIMARY_MOUSE = 1;
+
+export default class RegionCreator extends React.Component<Props, State> {
+    creatorRef: React.RefObject<HTMLDivElement> = React.createRef();
+
+    state: State = {
+        isDrawing: false,
+        position: null,
+    };
+
+    componentDidMount(): void {
+        document.addEventListener('mousemove', this.handleMouseMove);
```

Comment:
Should these be added to some reference element rather than document? `creatorRef` maybe?

Reply:
I had them on the creator ref originally, but it causes issues in edge cases where the cursor leaves the drawing area. I can add a comment explaining as much.